src/animation/AnimationClip.js

```diff
@@ -118,14 +42,14 @@ Object.assign( AnimationClip, {
 
 		}
 
-		const clip = new AnimationClip( json.name, json.duration, tracks, json.blendMode );
+		const clip = new this( json.name, json.duration, tracks, json.blendMode );
```

Comment:
Hah! I didn't know you could do `new this()` 🤓

Reply:
Wait, I see you do `new this.constructor` a bit further down the file. What's the difference?

Reply:
Yeah! If you are constructing from a static method you can use `this()`, otherwise you have to use `this.constructor()`

```js
class Foo {

  constructor(attributes) {
    Object.assign(this, attributes);
  }

  static create(attributes) {
    return new this(attributes);
  }

  clone() {
    return new this.constructor(this) ;
  }
}

class Bar extends Foo { }


const objects = [
 new Foo({ name: 'foo1' }),
 Foo.create({ name: 'foo2' }),
 new Foo({ name: 'foo3' }).clone(),
 new Bar({ name: 'bar1' }),
 Bar.create({ name: 'bar2' }),
 new Bar({ name: 'bar3' }).clone(),
]

for (const obj of objects) {
  console.log(obj);
}
```

Outputs -->
```
Foo { name: 'foo1' }
Foo { name: 'foo2' }
Foo { name: 'foo3' }
Bar { name: 'bar1' }
Bar { name: 'bar2' }
Bar { name: 'bar3' }
```

Reply:
Dang. You know your stuff! 🙏